src/Task/PhpArkitect.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+declare(strict_types=1);
+
+namespace GrumPHP\Task;
+
+use GrumPHP\Runner\TaskResult;
+use GrumPHP\Runner\TaskResultInterface;
+use GrumPHP\Task\Context\ContextInterface;
+use GrumPHP\Task\Context\GitPreCommitContext;
+use GrumPHP\Task\Context\RunContext;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+/**
+ * PHPArkitect task.
+ */
+class PhpArkitect extends AbstractExternalTask
+{
+    public static function getConfigurableOptions(): OptionsResolver
+    {
+        $resolver = new OptionsResolver();
+        $resolver->setDefaults([
+            'config' => null,
+            "target_php_version" => null,
+            'stop_on_failure' => null,
+        ]);
+
+        $resolver->addAllowedTypes('config', ['null', 'string']);
+        $resolver->addAllowedTypes('target_php_version', ['null', 'string']);
+        $resolver->addAllowedTypes('stop_on_failure', ['null', 'boolean']);
+
+        return $resolver;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function canRunInContext(ContextInterface $context): bool
+    {
+        return $context instanceof GitPreCommitContext || $context instanceof RunContext;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function run(ContextInterface $context): TaskResultInterface
+    {
+        $config = $this->getConfig()->getOptions();
+
+        $arguments = $this->processBuilder->createArgumentsForCommand('phparkitect');
+        $arguments->add('check');
+        $arguments->addOptionalArgument('--config=%s', $config['config']);
+        $arguments->addOptionalArgument('--target-php-version=%s', $config['target_php_version']);
+        $arguments->addOptionalArgument('--stop-on-failure', $config['stop_on_failure']);
+        $arguments->add('--no-ansi');
```

Comment:
Would it make sense not to provide this argument and let the system figure out if it needs to use ansi or not?
(the tool is based on symfony/console, so it should be able to detect and display ansi as wel)

Reply:
I based this class on the PhpStan task. The `--no-ansi` is added there by default. I wasn't aware symfony/console detects this.